### DISCUSSION TITLE
Why does brew warn about things in /usr/local/ when it's installed to /opt/homebrew/ ?

### DISCUSSION THREAD

**Discussion by tjluoma** [Homebrew/discussions#1959] [Getting started]
I recently installed [wkhtmltopdf](https://wkhtmltopdf.org/) via `brew install wkhtmltopdf` and now `brew doctor` is warning me:

```
Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libwkhtmltox.0.12.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/wkhtmltox/image.h
  /usr/local/include/wkhtmltox/pdf.h
```

This is on an M1-based Mac, where `brew` installs to `/opt/homebrew/` not `/usr/local/` so why is `brew doctor` complaining about files in `/usr/local/`? I thought one of the advantages of using the new directory was not having to worry about these sorts of potential collisions.

**Comment by gromgit** [comment]
As the warnings say:
> they could cause problems when building Homebrew formulae

`/usr/local` is the _de facto_ root directory for non-system software installations across most of Unixdom (including macOS), and most build configuration tools will automatically look in `/usr/local` to satisfy source dependencies. Homebrew's M1 compiler shims will generally filter out these paths _during compilation_, but all other tools are not similarly protected.

In particular, if a formula requires another language formula like `go` or `erlang` to build, Homebrew can't stop the former's build configuration from finding the wrong stuff in `/usr/local`, not can it stop the latter from using it.
